File: sample/impl/rcode/transforms/dummy/make_dummy_list.R
# %%%
# name: MakeDummyList
# type: make_dummy_list
# affinity: cpu
# version: 0.0.1
# inputs:
# 	cut_str:
# outputs:
# 	str_list:
# desc:
# %%%


source("core/executors/r/transform_base.R")

MakeDummyList <- R6Class(
  classname = "MakeDummyList",
  inherit = TransformBase,
  public = list(
    apply = function() {
      cut_str <- self$inputs[["cut_str"]][["value"]]
      n_repeat <- self$params[["repeat"]]
      str_list <- rep(cut_str, each=n_repeat)
      self$outputs <- list(
        "str_list" = list("value"=str_list)
      )
    }
  )
)

File: sample/impl/pycode/transforms/dummy/make_dummy_list.py
# %%%
# name: MakeDummyList
# type: make_dummy_list
# affinity: cpu
# version: 0.0.1
# inputs:
# 	cut_str:
# outputs:
# 	str_list:
# desc:
# %%%

from molecule.executors.py.transform_base import TransformBase, Dataset

class MakeDummyList(TransformBase):
  def apply(self):
    cut_str = self.inputs['cut_str'].value
    repeat = self.params['repeat']
    str_list = [cut_str] * repeat
    self.logger.info('test logging')
    self.outputs['str_list'] = Dataset(value=str_list)
